graphql_compiler/tests/snapshot_tests/test_query_pagination.py

```diff
@@ -1182,3 +1184,45 @@ def test_impossible_pagination(self):
         compare_graphql(self, original_query.query_string, first.query_string)
         self.assertEqual(original_query.parameters, first.parameters)
         self.assertEqual(0, len(remainder))
+
+    @pytest.mark.usefixtures("snapshot_orientdb_client")
+    def test_with_compiler_tests(self):
+        """Test that pagination doesn't crash on any of the queries from the compiler tests."""
+        schema_graph = generate_schema_graph(self.orientdb_client)
+        graphql_schema, type_equivalence_hints = get_graphql_schema_from_schema_graph(schema_graph)
+        pagination_keys = {vertex_name: "uuid" for vertex_name in schema_graph.vertex_class_names}
+        uuid4_fields = {vertex_name: {"uuid"} for vertex_name in schema_graph.vertex_class_names}
+        count_data = {vertex_name: 100 for vertex_name in schema_graph.vertex_class_names}
+        count_data.update({edge_name: 100 for edge_name in schema_graph.edge_class_names})
+        statistics = LocalStatistics(count_data)
+        schema_info = QueryPlanningSchemaInfo(
+            schema=graphql_schema,
+            type_equivalence_hints=type_equivalence_hints,
+            schema_graph=schema_graph,
+            statistics=statistics,
+            pagination_keys=pagination_keys,
+            uuid4_fields=uuid4_fields,
+        )
+
+        arbitrary_value_for_type = {
+            "String": "lol",
+            "ID": "40000000-0000-0000-0000-000000000000",
+            "Int": 5,
+            "Date": datetime.date(2000, 1, 1),
+            "DateTime": datetime.datetime(2000, 1, 1),
+            "Decimal": 5.3,
+            "[String]": ["lol", "lel"],
+        }
+
+        for test_name in dir(test_input_data):
+            method = getattr(test_input_data, test_name)
+            if hasattr(method, "__annotations__"):
```

Comment:
Use `get_test_function_names_from_class` from `test_testing_invariants.py` for this — let's have one way to do this. Maybe also move it to the `helpers` file as well.